### PR TITLE
Update ListContainer.java to make nested properties sortable

### PR DESCRIPTION
**Pull request by AdrienC**

### CODE REVIEW

**Comment by mstahv** [comment]
Looks pretty good to me, do you have some test case for this? Would be awesome to have some UI level test or a unit test for this.

**Comment by dve** [comment]
You can see the effect in `org.vaadin.viritin.it.MTableWithNestedProperty` without the patch the nested properties are not sortable, with the patch they are.

File: src/main/java/org/vaadin/viritin/ListContainer.java [hunk 1]
```diff
@@ -482,15 +485,16 @@ private PropertyComparator(Object[] propertyId, boolean[] ascending) {
...
                 String currentProperty = propertyId[i].toString();
-                Comparator<T> currentComparator
-                        = new BeanComparator<T>(currentProperty,
-                                getUnderlyingComparator(currentProperty));
+                Comparator underlyingComparator = getUnderlyingComparator(currentProperty);
+                Comparator currentComparator = underlyingComparator != null ? underlyingComparator : ComparableComparator.getInstance();
 
                 if (!ascending[i]) {
-                    currentComparator = new ReverseComparator(currentComparator);
+                    currentComparator = currentComparator.reversed();
```
Comment (mstahv) [inline src/main/java/org/vaadin/viritin/ListContainer.java]: reversed is default method from JDK8? Can you change this back in to using ReverseComparator as Viritin still supports JDK7.

**Comment by AdrienC** [comment]
Sorry..

File: src/main/java/org/vaadin/viritin/ListContainer.java [hunk 2]
```diff
@@ -482,15 +485,16 @@ private PropertyComparator(Object[] propertyId, boolean[] ascending) {
...
                 }
 
-                int compare = currentComparator.compare(o1, o2);
+                Object o1Prop = getContainerProperty(o1, currentProperty).getValue();
```
Comment (mstahv) [inline src/main/java/org/vaadin/viritin/ListContainer.java]: Not sure, but should check if this is in par with the old version performance vice. Also I'd more rely on BeanUtils API than on the Vaadin's Property API. Have you tried if BeanComparator supports nested property? I think the actual thing we are trying to achieve with this patch doesn't require this change.

**Comment by mstahv** [comment]
Referring my previous comment to outdated patch, maybe only the upper part of of the change is actually needed? If the other change is obsolete, I'd commit only the first change. Then commit the other if it helps in some other way.

**Comment by AdrienC** [comment]
BeanComparator supports nested properties, but this change makes generated columns sortable in MGrid.

**Comment by mstahv** [comment]
Thanks, finally had the time and courage to push this in. I'm trying to push in lot of container related stuff now, I hope you guys help me testing the stuff.

**Pull request merged**